presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -95,6 +100,11 @@ public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
             if (simplified.equals(BooleanLiteral.TRUE_LITERAL)) {
                 return source;
             }
+            // TODO: this needs to check whether the boolean expression coerces to false in a more general way.
+            // E.g., simplify() not always produces a literal when the expression is constant
+            else if (BooleanLiteral.FALSE_LITERAL.equals(simplified) || simplified instanceof NullLiteral) {
```

Comment:
Alternatively, you can update condition in first ```if```, to match your ```else if``` condition. Then code these conditions will be coherent (easier to understand) and code will look like anywhere else in the code. Up to you.

What about static imports for TRUE_LITERAL and FALSE_LITERAL. I would use them. Would you?